src/index/IndexImpl.Text.cpp

```diff
@@ -107,14 +107,14 @@ void IndexImpl::addTextFromContextFile(const string& contextFile,
   LOG(DEBUG) << "Reloading the RDF vocabulary ..." << std::endl;
   vocab_ = RdfsVocabulary{};
   readConfiguration();
-  vocab_.readFromFile(onDiskBase_ + INTERNAL_VOCAB_SUFFIX,
-                      onDiskBase_ + EXTERNAL_VOCAB_SUFFIX);
+  vocab_.readFromFile(onDiskBaseVocabulary_ + INTERNAL_VOCAB_SUFFIX,
```

Comment:
If you make the `onDiskBaseVocabulary_` an `optional` (see my suggestion above), and then instead introduce the function `onDiskBaseVocabulary()` that returns something like `onDiskBaseVocabulary_.value_or(onDiskBaseIndex_)`, then everything is clean nice and typesafe:)